### PR TITLE
Add locally built docker compose & docker usage instructions

### PR DESCRIPTION
**Pull request by ItsDrike**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,6 +23,25 @@ You'll now need to configure the bot. See the [configuring section](#configuring
 
 Finally, you can start the bot with `python -m src`.
 
+### Using docker
```
Comment (Paillat-dev) [inline README.md:26]: Might be beneficial to have some example commands.

File: README.md [hunk 2]
```diff
@@ -23,6 +23,33 @@ You'll now need to configure the bot. See the [configuring section](#configuring
...
+
+To use docker, you can check out the images that are automatically built after each update to the `main` branch on
+[ghcr](https://github.com/itsdrike/code-jam-2024/pkgs/container/code-jam-2024). You can also use [`docker
+compose`](https://docs.docker.com/compose/) with the [`docker-compose.yaml`](./docker-compose.yaml) file, which will
+pull this image from ghcr. To run the container using this file, you can use the following command:
+
+```bash
+docker compose up
+```
+
```
Comment (Paillat-dev) [inline README.md:40]:
```suggestion

> [!NOTE]  
> To run the container in the background, add the `-d` flag to the command.

```

File: README.md [hunk 3]
```diff
@@ -23,6 +23,33 @@ You'll now need to configure the bot. See the [configuring section](#configuring
 
 Finally, you can start the bot with `python -m src`.
 
+### Using docker
+
+The project also supports [docker](https://www.docker.com/) installation, which should allow running the project
+anywhere, without installing all of the dependencies manually. This is a lot more convenient way to run the bot, if you
+just want to run it and you don't wish to do any actual development.
+
+To use docker, you can check out the images that are automatically built after each update to the `main` branch on
```
Comment (Paillat-dev) [inline README.md]:
```suggestion
To use docker, you can check out the images that are automatically built after each update to the `main` branch at
```

File: README.md [hunk 4]
```diff
@@ -23,6 +23,33 @@ You'll now need to configure the bot. See the [configuring section](#configuring
...
+compose`](https://docs.docker.com/compose/) with the [`docker-compose.yaml`](./docker-compose.yaml) file, which will
+pull this image from ghcr. To run the container using this file, you can use the following command:
+
+```bash
+docker compose up
+```
+
+If you want to build the image locally (to include some other changes that aren't yet in the main branch, maybe during
+development or to customize something when deploying), you can also use the
+[`docker-compose.local.yaml`](./docker-compose.local.yaml), which defines an image building step from our
```
Comment (Paillat-dev) [inline README.md:46]:
```suggestion
[`docker-compose.local.yaml`](./docker-compose.local.yaml) file, which defines an image building step from our
```

File: README.md [hunk 5]
```diff
@@ -23,6 +23,33 @@ You'll now need to configure the bot. See the [configuring section](#configuring
...
+[`Dockerfile`](./Dockerfile). To run this local version of docker-compose, you can use the following command:
+
+```bash
+docker compose -f ./docker-compose.local.yaml up
+```
+
+Note that you will still need to create a `.env` file with all of the configuration variables (see [the configuring
```
Comment (Paillat-dev) [inline README.md]:
```suggestion
> [!IMPORTANT]  
> You will still need to create a `.env` file with all of the configuration variables (see [the configuring section](#configuring-the-bot).
```

**Comment by Paillat-dev** [comment]
For reference: https://github.com/orgs/community/discussions/16925

File: README.md (hunk 4 above)
Comment (Paillat-dev) [reply README.md:46]: Alternatively remove "the" before the hyperlink.

**Comment by ItsDrike** [comment]
> For reference: https://github.com/orgs/community/discussions/16925

yeah, I'm quite aware, used it countless times

**Review by Paillat-dev — APPROVED** [review]

**Pull request merged**